### PR TITLE
[Display Layout] Composition and configuration sync

### LINKED ISSUE #4661: unable to remove object from composition
#### Summary
run master on local machine or use /latest
trying to remove and object from composition does not remove it.

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug?

#### Steps to Reproduce
1. use old folder with couple of items in it (eg: Display Layout, SWG, clock etc)
2. context click on SWG under folder and select `remove`
3. click `ok` (observe selected SWG still inside folder)
3. click on Display Layout, got to  edit mode
4. try to remove SWG from elements pool (observe same result)

#### Environment
* Open MCT Version: 
 ``` Version Information
Version: 1.8.3-SNAPSHOT
Build Date: Mon Jan 03 2022 15:50:49 GMT-0800 (Pacific Standard Time)
Revision: 3a65f75d2190c464712fd4ee5c47f7255423da13
Branch: master
```
* Deployment Type: 
* OS:
* Browser:

#### Additional Information
1. caused by, Rewrite local storage (#4583) 2d64813a4fa5bfb69f16f7ab8e2753f73f232a8b
2. only unable to delete old objects
3. can delete newly created objects

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes #4661

### Describe your changes:

If you removed an item from a display layout while viewing the display layout, it would also update it's configuration to remove any additional metadata (ie, position, etc). If you were to remove an item from a display layout when not viewing the display layout, this step is never completed and the display layout still thinks it has an item to display (despite it not being in the composition). This fix will sync display layout composition and configuration on loading to prevent this from happening.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate unit tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Commit messages meet standards?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by unlikelyzero** [comment]
@jvigliotta can you create a super quick e2e test to cover this scenario?

**Comment by jvigliotta** [comment]
> @jvigliotta can you create a super quick e2e test to cover this scenario? a 30-minutes-or-the-pizzas-free version?

Done did it.

File: e2e/tests/functional/plugins/displayLayout/displayLayout.e2e.spec.js [hunk 1]
```diff
@@ -93,6 +93,70 @@ test.describe('Testing Display Layout @unstable', () => {
...
+        await page.locator('.c-tree__item.is-navigated-object .c-disclosure-triangle').click();
+
+        // Bring up context menu and remove
+        await page.locator('.c-tree__item.is-alias .c-tree__item__name:text("Test Sine Wave Generator")').first().click({ button: 'right' });
+        await page.locator('text=Remove').click();
+        await page.locator('text=OK').click();
+
+        // delete
+
+        expect.soft(await page.locator('.l-layout .l-layout__frame').count()).toEqual(0);
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/displayLayout/displayLayout.e2e.spec.js:124]: you don't have to use softs on the last expect, but it doesn't hurt

Comment (jvigliotta) [reply e2e/tests/functional/plugins/displayLayout/displayLayout.e2e.spec.js:124]: ahh, good to know!

**Review by davetsay — APPROVED** [review]

**Pull request merged**